pushflatpakscript/src/pushflatpakscript/flathub.py

```diff
@@ -138,6 +138,23 @@ def files(tar, path):
     return os.path.join(flatpak_tar_basedir, flatpak_deflated_dir)
 
 
+def check_app_id_matches_flatpak(context, flatpak_path):
+    # Extract all ostree refs from the supplied Flatpak repo
+    flatpak_refs = subprocess.check_output(["ostree", "refs"], cwd=flatpak_path).decode().splitlines()
+    
+    # Consolidate ostree refs into list of Flatpak IDs available in repo
+    flatpak_refs = [ref.split("/")[1] for ref in flatpak_refs if ref.startswith("app/")]
+    
+    # Create a list, if any, of all unexpected Flatpak IDs present in repo
+    invalid_refs = [ref for ref in flatpak_refs if not context.config["app_id"]]
```

Comment:
That check looks broken, presumably what you want is something like `invalid_refs = set(flatpak_refs) - {context.config["app_id"]}`.

If possible we should have a test to cover that case.

Reply:
I've updated and tested the patch with your suggested change. Much more elegant than what I was trying to do!

Reply:
If possible I'd like to write the test in a separate PR so this can land and we can test our Flatpak pushing before Flathub implements their breaking changes in a few weeks.